web/components/inputs/FcInputLocationSearch.vue

```diff
@@ -169,6 +166,29 @@ export default {
       this.locationSuggestions = [];
       this.query = null;
       this.state = LocationSearchState.VALUE_EMPTY;
+      if (this.hasLocationIndex && !this.hasLocationToAddIndex) {
+        this.$emit('location-remove', this.locationIndex);
+      }
+    },
+    actionBlur(e) {
+      /*
+       * Clicking a location suggestion in the dropdown triggers a blur event.  However, if
+       * we set `this.hasFocus = false` immediately, the dropdown disappears before the `@click`
+       * handler is processed.
+       *
+       * By detecting this case here, we allow the state transition to
+       * `LocationSearchState.VALUE_SELECTED` within `actionSelect` to hide the dropdown instead,
+       * after the `@click` handler is processed.
+       */
+      const { relatedTarget: $relatedTarget } = e;
+      if ($relatedTarget !== null) {
+        const $list = $relatedTarget.closest('.fc-list-location-suggestions');
+        if (this.$refs.listLocationSuggestions !== undefined
```

Comment:
add comment here explaining need for `undefined` check, with [link to docs](https://vuejs.org/v2/guide/instance.html#Lifecycle-Diagram)